3567/0/about_scoring_project.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+require File.expand_path(File.dirname(__FILE__) + '/neo')
+
+# Greed is a dice game where you roll up to five dice to accumulate
+# points.  The following "score" function will be used to calculate the
+# score of a single roll of the dice.
+#
+# A greed roll is scored as follows:
+#
+# * A set of three ones is 1000 points
+#
+# * A set of three numbers (other than ones) is worth 100 times the
+#   number. (e.g. three fives is 500 points).
+#
+# * A one (that is not part of a set of three) is worth 100 points.
+#
+# * A five (that is not part of a set of three) is worth 50 points.
+#
+# * Everything else is worth 0 points.
+#
+#
+# Examples:
+#
+# score([1,1,1,5,1]) => 1150 points
+# score([2,3,4,6,2]) => 0 points
+# score([3,4,5,3,3]) => 350 points
+# score([1,5,1,2,4]) => 250 points
+#
+# More scoring examples are given in the tests below:
+#
+# Your goal is to write the score method.
+# rubocop:disable Metrics/MethodLength
+def score(dice)
+  # rubocop:enable Metrics/MethodLength
+  score = 0
+  
+  count_1 = dice.find_all { |i| i == 1 }.count
+  count_2 = dice.find_all { |i| i == 2 }.count
+  count_3 = dice.find_all { |i| i == 3 }.count
+  count_4 = dice.find_all { |i| i == 4 }.count
+  count_5 = dice.find_all { |i| i == 5 }.count
+  count_6 = dice.find_all { |i| i == 6 }.count
+ 
+  score += 1000 * (count_1 / 3)
+  score += 200 * (count_2 / 3)
+  score += 300 * (count_3 / 3)
+  score += 400 * (count_4 / 3)
+  score += 500 * (count_5 / 3)
+  score += 600 * (count_6 / 3)
+  score += 100 * (count_1 % 3)
+  score += 50  * (count_5 % 3)
```

Comment:
It's simple math!
Tell me what would you do if we have dice with 18 sides? 100?